src/theory/sets/cardinality_extension.cpp

```diff
@@ -121,22 +126,26 @@ void CardinalityExtension::checkFiniteType(TypeNode& t)
   // get all equivalent classes of type t
   vector<Node> representatives = d_state.getSetsEqClasses(t);
 
-  Node typeCardinality = nm->mkConst(Rational(card.getFiniteCardinality()));
-
-  Node cardUniv = nm->mkNode(kind::CARD, proxy);
-  Node leq = nm->mkNode(kind::LEQ, cardUniv, typeCardinality);
-
-  // (=> true (<= (card (as univset t)) cardUniv)
-  if (!d_state.isEntailed(leq, true))
+  if (t.isInterpretedFinite())
   {
-    d_im.assertInference(leq, d_true, "finite type cardinality", 1);
+    Node typeCardinality = nm->mkConst(Rational(card.getFiniteCardinality()));
+    Node cardUniv = nm->mkNode(kind::CARD, proxy);
+    Node leq = nm->mkNode(kind::LEQ, cardUniv, typeCardinality);
+
+    // (=> true (<= (card (as univset t)) cardUniv)
+    if (!d_state.isEntailed(leq, true))
+    {
+      d_im.assertInference(
+          leq, d_true, "univset cardinality <= type cardinality", 1);
+    }
   }
 
-  // add subset lemmas for sets, and membership lemmas for negative members
+  // add subset lemmas for sets and membership lemmas for negative members
   for (Node& representative : representatives)
   {
-    // the universe set is a subset of itself
-    if (representative != d_ee.getRepresentative(univ))
+    if (representative
+        != d_ee.getRepresentative(
+            univ))  // the universe set is a subset of itself
```

Comment:
Nit: can you put the comment on the previous line so that this condition is not ugly after clang-format?